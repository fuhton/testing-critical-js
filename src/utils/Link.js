import React from "react";

export default ({ className = "default-class-name", children, page }) => (
  <a className={className} href={page}>{children}</a>
);