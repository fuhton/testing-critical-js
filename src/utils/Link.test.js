import React from "react";
import Link from "./Link";
import renderer from "react-test-renderer";

describe("utils/Link", () => {
  describe("Link", () => {
    it("should render with default props", () => {
      const link = renderer
        .create(<Link page="a.new.url">A New Url</Link>)
        .toJSON();
      expect(link).toMatchSnapshot();
    });
    it("should render with passed class name", () => {
      const link = renderer
        .create(
          <Link className="test" page="a.new.url">
            A New Url
          </Link>
        )
        .toJSON();
      expect(link).toMatchSnapshot();
    });
  });
});
