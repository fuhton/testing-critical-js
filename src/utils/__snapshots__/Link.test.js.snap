// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/Link Link should render with default props 1`] = `
<a
  className="default-class-name"
  href="a.new.url"
>
  A New Url
</a>
`;

exports[`utils/Link Link should render with passed class name 1`] = `
<a
  className="test"
  href="a.new.url"
>
  A New Url
</a>
`;
